### PR TITLE
fix job log reporting with gradual progress completion

### LINKED ISSUE #90: A job's 'percentCompleted' is not the same percent as in the status message
During a WPS1 execution, we can see the percentage at 2 places: in the `percentCompleted` field, and in the `message` field. The value in the message field seems to be the good one.

At one point, the percent shown in the message was 90% (right at the end of the processing, so `REMOTE_JOB_PROGRESS_FETCH_OUT` was the right value), and in the `percentCompleted` field, it was 30%.

After a quick look at the code, I think that the `weaver.process.utils.map_progress` function is called twice (in 2 different places) before being outputted in `percentCompleted`, but once for the value in the status message.

So this one is ok:
https://github.com/crim-ca/weaver/blob/339e57b7a03a82e840c6068ce5919b131afef48f/weaver/processes/wps1_process.py#L132

But this one should treat the progress value as the final percentage I believe (or at least in this case):
https://github.com/crim-ca/weaver/blob/339e57b7a03a82e840c6068ce5919b131afef48f/weaver/processes/wps_package.py#L1709

### PR DESCRIPTION
**Pull request by fmigneault**
fixes #90

Job execution logs side-by-side comparison of before/after changes. 
Instead of reporting same (final) percent all the time and status `failed`/`succeeded` accordingly to final result, reports `running` (which was the status of the job at the time they were logged) and make percent proportionally increase. 

Sadly, reported progress/status is still limited to how verbose the package is *during* its execution - and collected after the fact to apply to the job log, but they are more consistent with what happened otherwise. 

image

### CODE REVIEW

File: weaver/processes/execution.py [hunk 1]
```diff
@@ -201,20 +204,21 @@ def execute_process(self, job_id, url, headers=None):
...
                 if execution.isComplete():
                     job.mark_finished()
-                    job.progress = JOB_PROGRESS_EXECUTE_MONITOR_END
                     msg_progress = " (status: {})".format(job_msg) if job_msg else ""
                     if execution.isSucceded():
+                        wps_package.retrieve_package_job_log(execution, job, progress_min, progress_max)
                         job.status = map_status(STATUS_SUCCEEDED)
                         job.status_message = "Job succeeded{}.".format(msg_progress)
-                        wps_package.retrieve_package_job_log(execution, job)
+                        job.progress = JOB_PROGRESS_EXECUTE_MONITOR_DONE
```
Comment (MatProv) [inline weaver/processes/execution.py:212]: Why not use `progress_max`? Same comment for below usage of `JOB_PROGRESS_EXECUTE_MONITOR_DONE `.

File: weaver/processes/wps_package.py [hunk 2]
```diff
@@ -164,8 +164,13 @@ def retrieve_package_job_log(execution, job):
         # if the process is a weaver package this status xml should be available in the process output dir
         log_path = get_status_location_log_path(execution.statusLocation, out_dir=out_dir)
         with open(log_path, "r") as log_file:
-            for line in log_file:
-                job.save_log(message=line.rstrip("\n"))
+            log_lines = log_file.readlines()
+        if not log_lines:
+            return
```
Comment (MatProv) [inline weaver/processes/wps_package.py:169]: Simply return or throw exception?

File: weaver/processes/wps_package.py [hunk 3]
```diff
@@ -164,8 +164,13 @@ def retrieve_package_job_log(execution, job):
...
+        total = float(len(log_lines))
+        for i, line in enumerate(log_lines):
+            progress = map_progress(i / total * 100, progress_min, progress_max)
```
Comment (MatProv) [inline weaver/processes/wps_package.py:172]: What if `len(log_lines) == 0`?

**Review by MatProv — APPROVED** [review]

File: weaver/processes/wps_package.py (hunk 2 above)
Comment (fmigneault) [reply weaver/processes/wps_package.py:169]: Simply return. If the process was not verbose, we won't break its execution for it. Logs will just be minimal.

**Review by dbyrns — COMMENTED** [review]
I wanted to let you know that I still validate and saw nothing wrong.

**Pull request merged**